app/models/reducers/count_choices_reducer.rb

```diff
@@ -0,0 +1,13 @@
+module Reducers
+  class CountChoicesReducer < Reducer
+    attr_reader :sub_ranges
+
+    def reduction_data_for(extractions)
+      ReductionResults.build do |results|
+        extractions.each do |extraction|
```

Comment:
perhaps encapsulate the nested loop in a method? I find multi-nested loops hard to interpret, especially if we are yielding params in there.